src/configure/configure.ts

```diff
@@ -177,73 +181,131 @@ class PipelineConfigurer {
-                            {
-                                openLabel: Messages.selectLabel,
-                                canSelectFiles: false,
-                                canSelectFolders: true,
-                                canSelectMany: false
-                            }
-                        );
-                        if (selectedFolder && selectedFolder.length > 0) {
-                            this.workspacePath = selectedFolder[0].fsPath;
-                        }
-                        else {
-                            throw new Error(Messages.noWorkSpaceSelectedError);
-                        }
-                        break;
-                    case SourceOptions.CurrentWorkspace:
-                        this.workspacePath = vscode.workspace.rootPath;
-                        break;
-                    default:
-                        exit(0);
-                }
+                await this.setWorkspace();
             }
 
-            await this.getGitDetailsFromRepository(this.workspacePath);
+            await this.getGitDetailsFromRepository();
         }
         catch (error) {
             this.telemetryHelper.logError(Layer, TracePoints.GetSourceRepositoryDetailsFailed, error);
             throw error;
         }
     }
 
-    private async getGitDetailsFromRepository(workspacePath: string): Promise<void> {
-        this.localGitRepoHelper = await LocalGitRepoHelper.GetHelperInstance(workspacePath);
-        this.inputs.sourceRepository = await this.localGitRepoHelper.getGitRepoDetails(workspacePath);
+    private async setWorkspace(): Promise<void> {
+        let workspaceFolders = vscode.workspace && vscode.workspace.workspaceFolders;
+        if (workspaceFolders && workspaceFolders.length > 0) {
+            this.telemetryHelper.setTelemetry(TelemetryKeys.SourceRepoLocation, Messages.selectWorkspaceFolder);
+
+            if (workspaceFolders.length === 1) {
+                this.workspacePath = workspaceFolders[0].uri.fsPath;
+            }
+            else {
+                let workspaceFolderOptions: Array<QuickPickItemWithData> = [];
+                for (let folder of workspaceFolders) {
+                    workspaceFolderOptions.push({ label: folder.name, data: folder });
+                }
+                let selectedWorkspaceFolder = await this.controlProvider.showQuickPick(
+                    workspaceFolderOptions,
+                    { placeHolder: Messages.selectWorkspaceFolder });
+                this.workspacePath = selectedWorkspaceFolder.data.uri.fsPath;
+            }
+        }
+        else {
+            this.telemetryHelper.setTelemetry(TelemetryKeys.SourceRepoLocation, Messages.selectLabel);
+            let selectedFolder: vscode.Uri[] = await vscode.window.showOpenDialog(
+                {
+                    openLabel: Messages.selectLabel,
+                    canSelectFiles: false,
```

Comment:
should we move this to controlProvider ? along with telemetry above

Reply:
There is no telemetry to set while this control is displayed unlike dropdowns, so I used the vscode control as it is without overriding.

Reply:
The telemetry I have added is not generic or tied to the control, it is very specific to current flow so leaving as it is

Reply:
Yeah makes sense